libs/mpi/tests/unit/mpi_ring_async_executor.cpp

```diff
@@ -94,7 +94,7 @@ int hpx_main(hpx::program_options::variables_map& vm)
                 {
                     // send the incremented token to the next rank
                     ++tokens[i];
-                    hpx::future<void> f_send = hpx::async(
+                    hpx::future<int> f_send = hpx::async(
```

Comment:
The return value is never used, why should it be explicit?

Reply:
As a side note: any exceptions possibly thrown from the `hpx::async` are not propagated by this code and are silently swallowed. I have missed this when working on the code as well. Would you mind adding the proper exception propagation while you are at it?

Reply:
I added the return value for completeness - though we don't use it, but I could see someone complaining that if the call to mpi_isend failed (for whatever reason) then they should be able to retrieve the return value. I have not done exception handling yet (I presume you mean for mpi_calls that fail) and it's not high on my list. I will add it at some point, but I want to concentrate on fixing app related issues first.

Reply:
I was referring to any exception that might be thrown by `hpx::async`. All of those should end up in the returned future, but the code never calls `.get()` on it, so the exception will not rethrown.

Reply:
I want to make the code exception safe, but I'd rather do this in another PR as I am in a hurry to get more functionality working on other branches that are more important than this. I'll create an issue for it once this PR is merged (if ever).